plugins/modules/clustering/consul/consul.py

```diff
@@ -589,15 +573,27 @@ def main():
             token=dict(no_log=True)
         ),
         supports_check_mode=False,
+        mutually_exclusive=[
+            ('script', 'ttl', 'tcp', 'http'),
+        ],
+        required_by={
+            'script': 'interval',
+            'http': 'interval',
+            'tcp': 'interval',
+        },
+        required_if=[
+            ('state', 'present', ['service_name']),
+            ('state', 'absent', ['service_id', 'service_name', 'check_id', 'check_name']),
```

Comment:
At least one of these is required, but not all of them at once.
```suggestion
            ('state', 'absent', ['service_id', 'service_name', 'check_id', 'check_name'], True),
```

Reply:
Yup, I messed it up in mymind and thought it was the other way around. Thanks for spotting!